src/index/index_field.h

```diff
@@ -31,10 +31,13 @@ typedef struct {
 	AttributeID id;          // field id
 	IndexFieldType type;     // field type(s)
 	struct {
-		double weight;       // the importance of text
-		bool nostem;         // disable stemming of the text
-		char *phonetic;      // phonetic search of text
-		uint32_t dimension;  // vector dimension
+		double weight;         // the importance of text
+		bool nostem;           // disable stemming of the text
+		char *phonetic;        // phonetic search of text
+		uint32_t dimension;    // vector dimension
```

Comment:
would you consider grouping vector field options within an inner struct? e.g. HNSW params?